### DISCUSSION TITLE
Need information

### DISCUSSION THREAD

**Discussion by Mduroi** [fzyzcjy/flutter_rust_bridge#299] [Q&A]
Hello,

I start learning Dart and Rust later in my studies, and I came across this awesome project.

But I have a question that bothers me, if I have a Rust code that uses multithreading and async, can we call it from Dart knowing that it is single threaded ? 

If yes, could you explain to me the mechanisms that make it work (In a simple way).

Thanks.

**Comment by fzyzcjy** [comment]
> can we call it from Dart knowing that it is single threaded 

Sure.

> multithreading

By default when you use this lib to call rust from dart, the default handler uses a thread pool. so it is indeed multithreaded.

> async

https://github.com/fzyzcjy/flutter_rust_bridge/blob/master/book/src/async_in_rust.md by @AlienKevin
